tests/unit/s2n_policy_writer_test.c

```diff
@@ -0,0 +1,229 @@
+ */
+
+#include <fcntl.h>
+#include <string.h>
+#include <unistd.h>
+
+#include "s2n_test.h"
+#include "testlib/s2n_testlib.h"
+#include "tls/policy/s2n_policy_feature.h"
+
+DEFINE_POINTER_CLEANUP_FUNC(char *, unlink);
+
+static S2N_RESULT s2n_verify_format_v1_output(const char *output)
+{
+    RESULT_ENSURE_REF(output);
+
+    /* Required sections are present */
+    RESULT_ENSURE(strstr(output, "min version: ") != NULL, S2N_ERR_TEST_ASSERTION);
+    RESULT_ENSURE(strstr(output, "rules:\n") != NULL, S2N_ERR_TEST_ASSERTION);
+    RESULT_ENSURE(strstr(output, "cipher suites:\n") != NULL, S2N_ERR_TEST_ASSERTION);
+    RESULT_ENSURE(strstr(output, "signature schemes:\n") != NULL, S2N_ERR_TEST_ASSERTION);
+    RESULT_ENSURE(strstr(output, "curves:\n") != NULL, S2N_ERR_TEST_ASSERTION);
+
+    return S2N_RESULT_OK;
+}
+
+int main(int argc, char **argv)
+{
+    BEGIN_TEST();
+
+    /* Test: s2n_security_policy_write_length */
+    {
+        /* Test: safety - NULL policy */
+        {
+            uint32_t length;
+            EXPECT_FAILURE_WITH_ERRNO(
+                    s2n_security_policy_write_length(NULL, S2N_POLICY_FORMAT_DEBUG_V1, &length),
+                    S2N_ERR_NULL);
+        };
+
+        /* Test: safety - NULL length pointer */
+        {
+            const struct s2n_security_policy *policy = NULL;
+            EXPECT_SUCCESS(s2n_find_security_policy_from_version("default", &policy));
+            EXPECT_NOT_NULL(policy);
+
+            EXPECT_FAILURE_WITH_ERRNO(
+                    s2n_security_policy_write_length(policy, S2N_POLICY_FORMAT_DEBUG_V1, NULL),
+                    S2N_ERR_NULL);
+        };
+
+        /* Test: safety - invalid format */
+        {
+            const struct s2n_security_policy *policy = NULL;
+            EXPECT_SUCCESS(s2n_find_security_policy_from_version("default", &policy));
+            EXPECT_NOT_NULL(policy);
+
+            uint32_t length;
+            EXPECT_FAILURE_WITH_ERRNO(
+                    s2n_security_policy_write_length(policy, 999, &length),
+                    S2N_ERR_INVALID_ARGUMENT);
+        };
+
+        /* Test: successful length calculation and consistency with write_bytes */
+        {
+            const struct s2n_security_policy *policy = NULL;
+            EXPECT_SUCCESS(s2n_find_security_policy_from_version("default", &policy));
+            EXPECT_NOT_NULL(policy);
+
+            uint32_t required_length = 0;
+            EXPECT_SUCCESS(s2n_security_policy_write_length(policy, S2N_POLICY_FORMAT_DEBUG_V1, &required_length));
+            EXPECT_TRUE(required_length > 0);
+
+            DEFER_CLEANUP(struct s2n_blob exact_buffer = { 0 }, s2n_free);
+            EXPECT_SUCCESS(s2n_alloc(&exact_buffer, required_length));
+            EXPECT_SUCCESS(s2n_security_policy_write_bytes(policy, S2N_POLICY_FORMAT_DEBUG_V1, exact_buffer.data, required_length));
+
+            EXPECT_OK(s2n_verify_format_v1_output((const char *) exact_buffer.data));
```

Comment:
If you want to call `s2n_verify_format_v1_output` here I think +1 allocated size is needed. Otherwise if one of the `strstr` fails to find a match for some reason I think it will scan off the end of the buffer?